### ISSUE TITLE
passing data from selected item to the select function

### ISSUE THREAD

**Issue by GordonJones** [mar10/jquery-ui-contextmenu#126]
In the select function, the "cmd" ui attribute gets its value from the selected menu item.
I would like to be able to pass additional data from the selected item to the select function. I had naively thought that "extraData" would be passed through as well, but it appears not to be the case.

I am using "setEntry" in the beforeOpen function to dynamically add a sub menu with items of the form:

` {title: "aName", cmd: "changeType", extraData: {typeName: "aName"}, addClass: "aName}`

In the select function I thought I would be able to access my extraData attribute with

`ui.extraData.typeName`

but extraData is empty.

Am I missing something, or is there another way of achieving this. I guess I could use an action not use the select function?

**Comment by mar10** [comment]
I didn't look into it, but you could try to pass `data: ` instead, which should add an data attribute to the generated element.
Or maybe it helps if you do `ui.extraData.typeName = ...` in the beforeOpen event

**Comment by GordonJones** [comment]
Errm I am not making progress.
I set `data` instead of `extraData` as you suggest, but there is no data attribute accessible from the `ui` object at `select` time. The `ui` object at the time of selection has 4 attributes `cmd, extraData, item and target`, but none of these carries my data across.
I cannot give the ui object the typeName at `beforeOpen`, because the end user hasn't made a selection yet. I cannot guess which she will select.

Unless you have any alternative, I think I have 2 choices:

1. Use a cmd value that is in effect a command plus the extra data (eg `cmd:"changeType-aTypeName"`, and I parse it in the select function.
or
2. Use an action function in the menu item

**Comment by mar10** [comment]
Could you set up a jsFiddle, Plunkr or CodePen, that shows your scenario?

**Comment by GordonJones** [comment]
Hello Martin - sorry to go quiet for a long time, but I've been on Grandad duty for a while, and I also bought a wreck of a boat which needs lots of work. So my JS project is down the priority list - a bit.

I don't know how to generate the fiddle and plunker and codePen examples but I have got around this issue by carrying 2 bits of information in the single command value.

Since my target is to run my app in Electron, I decided now is a good time to attempt to port it across. It has proven painful, and I have hit a problem with your ui-contextmenu, which I will raise as a separate issue.
